RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate ThreadMessagesView to Typescript

## Proposed changes

Migrate ThreadMessagesView to Typescript

## Issue(s)	


## How to test or reproduce

- Go to a room
- Click on the icon side of the search icon
- Then you should be moved to the ThreadMessagesView, that is what we migrated
- Besides the view, the dropdown filter and search were migrated too, so check them.
- Furthermore, if don't exist threads, create them.

## Screenshots

## Types of changes




## Checklist



## Further comments
